This patch helps to avoid problems when running this test using some
simulators.  argv is not always available when running under a sim.
Unfortunately, this test might not be sufficient for some sims.

This patch or something like it should go upstream.

2011-11-11  Kevin Buettner  <kevinb@redhat.com>

	* gdb.python/py-value.c (main): Don't dereference argv when
	argv is NULL or argc is 0.


Index: gdb/testsuite/gdb.python/py-value.c
===================================================================
RCS file: /cvs/cvsfiles/gnupro/gdb/testsuite/gdb.python/py-value.c,v
retrieving revision 1.3
diff -u -p -r1.3 py-value.c
--- gdb/testsuite/gdb.python/py-value.c	2 Jul 2011 02:08:31 -0000	1.3
+++ gdb/testsuite/gdb.python/py-value.c	12 Nov 2011 00:07:54 -0000
@@ -69,7 +69,7 @@ int func2 (int arg1, int arg2)
 int
 main (int argc, char *argv[])
 {
-  char *cp = argv[0]; /* Prevent gcc from optimizing argv[] out.  */
+  char *cp = (argc && argv) ? argv[0] : NULL; /* Prevent gcc from optimizing argv[] out.  */
   struct s s;
   union u u;
   PTR x = &s;
